web/src/common/labels.ts

```diff
@@ -0,0 +1,58 @@
+import { msg } from "@lit/localize";
+
+import { EventActions, IntentEnum, SeverityEnum } from "@goauthentik/api";
+
+/* Various tables in the API for which we need to supply labels */
+
+export const intentEnumToLabel = new Map<IntentEnum, string>([
+    [IntentEnum.Api, msg("API Access")],
+    [IntentEnum.AppPassword, msg("App password")],
+    [IntentEnum.Recovery, msg("Recovery")],
+    [IntentEnum.Verification, msg("Verification")],
+    [IntentEnum.UnknownDefaultOpenApi, msg("Unknown intent")],
+]);
+
+export const intentToLabel = (intent: IntentEnum) => intentEnumToLabel.get(intent);
+
+export const eventActionToLabel = new Map<EventActions | undefined, string>([
+    [EventActions.Login, msg("Login")],
+    [EventActions.LoginFailed, msg("Failed login")],
+    [EventActions.Logout, msg("Logout")],
+    [EventActions.UserWrite, msg("User was written to")],
+    [EventActions.SuspiciousRequest, msg("Suspicious request")],
+    [EventActions.PasswordSet, msg("Password set")],
+    [EventActions.SecretView, msg("Secret was viewed")],
+    [EventActions.SecretRotate, msg("Secret was rotated")],
+    [EventActions.InvitationUsed, msg("Invitation used")],
+    [EventActions.AuthorizeApplication, msg("Application authorized")],
+    [EventActions.SourceLinked, msg("Source linked")],
+    [EventActions.ImpersonationStarted, msg("Impersonation started")],
+    [EventActions.ImpersonationEnded, msg("Impersonation ended")],
+    [EventActions.FlowExecution, msg("Flow execution")],
+    // These are different: look closely.
+    [EventActions.PolicyExecution, msg("Policy execution")],
+    [EventActions.PolicyException, msg("Policy exception")],
+    [EventActions.PropertyMappingException, msg("Property Mapping exception")],
+    // These are different: look closely.
+    [EventActions.SystemTaskExecution, msg("System task execution")],
+    [EventActions.SystemTaskException, msg("System task exception")],
+    [EventActions.SystemException, msg("General system exception")],
+    [EventActions.ConfigurationError, msg("Configuration error")],
+    [EventActions.ModelCreated, msg("Model created")],
+    [EventActions.ModelUpdated, msg("Model updated")],
+    [EventActions.ModelDeleted, msg("Model deleted")],
+    [EventActions.EmailSent, msg("Email sent")],
+    [EventActions.UpdateAvailable, msg("Update available")],
+]);
+
+export const actionToLabel = (action?: EventActions): string =>
+    eventActionToLabel.get(action) ?? action ?? "";
```

Comment:
~~we should probably have a more visible fallback like we do for the severity below~~

events can also have a custom action (which always has `custom_` prefixed), so as a fallback we should probably just return the action

Reply:
That's what this does; only if _that_ fails does it return an empty string. It's the same logic as the original, just less, ah, "iffy."  See: [web/src/admin/events/utils.ts](https://github.com/goauthentik/authentik/pull/6852/files#diff-467f1f91e11f15e16fb5d4465813d492864eddbb23f532b84abc722ab710d8bd)

Reply:
oh yeah I misread that, LGTM

Reply:
Cool, thanks. Yeah, I just started taking the `??` operator seriously recently myself, so I'm still getting used to its semantics. Sorry for going all hypergraphia on ya, I just didn't want to be misunderstood.